seqr/utils/file_utils.py

```diff
@@ -87,22 +90,27 @@ def get_gs_file_list(gs_path, user=None, check_subfolders=True):
 
     if check_subfolders:
         # If a bucket is empty gsutil throws an error when running ls with ** instead of returning an empty list
-        subfolders = _run_gsutil_with_wait(command, gs_path, user, get_stdout=True)
+        subfolders = _run_gsutil_with_stdout(command, gs_path, user)
         if not subfolders:
             return []
         gs_path = f'{gs_path}/**'
 
-    all_lines = _run_gsutil_with_wait(command, gs_path, user, get_stdout=True)
+    all_lines = _run_gsutil_with_stdout(command, gs_path, user)
     return [line for line in all_lines if is_google_bucket_file_path(line)]
 
 
-def _run_gsutil_with_wait(command, gs_path, user=None, get_stdout=False):
-    if not is_google_bucket_file_path(gs_path):
-        raise Exception('A Google Storage path is expected.')
+def _run_gsutil_with_wait(command, gs_path, user=None):
     process = _run_gsutil_command(command, gs_path, user=user)
     if process.wait() != 0:
         errors = [line.decode('utf-8').strip() for line in process.stdout]
         raise Exception('Run command failed: ' + ' '.join(errors))
-    if get_stdout:
-        return [line.decode('utf-8').rstrip('\n') for line in process.stdout]
     return process
+
+
+def _run_gsutil_with_stdout(command, gs_path, user=None):
+    process = _run_gsutil_command(command, gs_path, user=user)
+    output, errs = process.communicate()
+    if errs:
+        errors = errs.decode('utf-8').strip().replace('\n', ' ')
+        raise Exception(f'Run command failed: {errors}')
```

Comment:
In the case of the path not existing or insufficient permissions, the `errs` is still None. The error message is in the output because we set `stderr=subprocess.STDOUT`. We have to use other ways to detect errors, and I don't know how yet.

Reply:
oh good cacth